src/ncp/ncp_base.hpp

```diff
@@ -746,6 +755,16 @@ class NcpBase
 
     uint64_t mLogTimestampBase; // Timestamp base used for logging
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_NCP_INFRA_IF_ENABLE
+    otError InfraIfAddAddress(const otIp6Address &aAddress);
+    bool    InfraIfContainsAddress(const otIp6Address &aAddress);
+
+    static constexpr uint8_t kMaxInfraIfAddrs = 10;
+    otIp6Address             mInfraIfAddrs[kMaxInfraIfAddrs];
+    uint8_t                  mInfraIfAddrCount = 0;
+    uint32_t                 mInfraIfIndex     = 0;
```

Comment:
Could we initialize these in the `NcpBase` constructor instead?

We generally prefer initializing member variables within the constructor in the OT repository. This approach offers a couple of advantages:
- Consistency: It keeps the initialization logic centralized and easier to follow.
- Maintainability: All initial values are set in one place, simplifying debugging and future modifications.
- Performance: Some older/embedded toolchains might generate less efficient code for in-place member initialization.